src/components/Dropdowns/AccessibleDropdown.tsx

```diff
@@ -0,0 +1,242 @@
+ */
+export default function AccessibleDropdown(inputProps: AccessibleDropdownProps): ReactElement {
+  const props = { ...defaultProps, ...inputProps } as Required<AccessibleDropdownProps>;
+
+  //// Handle clicking on the dropdowns ///////////////////////////////////
+
+  // Support tab navigation by forcing the dropdown to stay open when clicked.
+  const [forceOpen, setForceOpen] = useState(false);
+  const componentContainerRef = useRef<HTMLDivElement>(null);
+
+  // If open, close the dropdown when focus is lost.
+  // Note that the focus out event will fire even if the newly focused element is also
+  // inside the component, so we need to check if the new target is also a child element.
+  useEffect(() => {
+    if (!forceOpen) {
+      return;
+    }
+    const doesContainTarget = (target: EventTarget | null): boolean => {
+      return (
+        (target instanceof Element &&
+          componentContainerRef.current &&
+          componentContainerRef.current.contains(target)) ||
+        false
+      );
+    };
+    const handleFocusLoss = (event: FocusEvent): void => {
+      if (!doesContainTarget(event.relatedTarget)) {
+        setForceOpen(false);
+      }
+    };
+
+    componentContainerRef.current?.addEventListener("focusout", handleFocusLoss);
+    return () => {
+      componentContainerRef.current?.removeEventListener("focusout", handleFocusLoss);
+    };
+  }, [forceOpen]);
+
+  //// Handle rendering of buttons and dropdown contents ///////////////////
+  const defaultRenderButton = (props: AccessibleDropdownProps, isOpen: boolean, onClick: () => void): ReactElement => {
+    return (
+      <MainButton
+        disabled={props.disabled}
+        style={props.buttonStyle}
+        $type={props.buttonType}
+        type={props.buttonType === "outlined" ? "default" : props.buttonType}
+        $open={isOpen}
+        // Open the dropdown when clicked for accessibility
+        onClick={onClick}
+      >
+        <MainButtonContents>
+          <div>
+            {props.buttonText}
+            <VisuallyHidden>(click to open menu)</VisuallyHidden>
+          </div>
+          <DropdownSVG />
+        </MainButtonContents>
+      </MainButton>
+    );
+  };
+  const renderButton = props.renderButton || defaultRenderButton;
+
+  // Use Ant styling for default dropdown content rendering
+  const { token } = theme.useToken();
+  const defaultRenderDropdownContent = (content: ReactElement): ReactElement => {
+    // Fake the menu background styling
+    const dropdownStyle: React.CSSProperties = {
+      backgroundColor: token.colorBgElevated,
+      borderRadius: token.borderRadiusLG,
+      boxShadow: token.boxShadowSecondary,
+    };
+    return <DropdownContentWrapper style={dropdownStyle}>{content}</DropdownContentWrapper>;
+  };
+  const renderDropdownContent = props.renderDropdownContent || defaultRenderDropdownContent;
```

Comment:
Same question as above here, and I acknowledge this isn't part of the change set, but as it is a big PR I'm struck by the fact that there some of these optional props aren't used anywhere currently.

On the one hand I dig that thoroughness, and it has me thinking about building a shared set of components for use in different AICS projects, it could be really useful when trying to make Simularium more accessible if we could port accessible components from NucMorph over there and their generality is an asset in that case. But could also be a YAGNI situation?

Reply:
Definitely up for debate. I'm comfortable with leaving them in I think, because I do anticipate situations where custom components may want to adjust the rendering style.